helm/cluster-cloud-director/values.yaml

```diff
@@ -101,3 +101,5 @@ kubectlImage:
   registry: quay.io
   name: giantswarm/kubectl
   tag: 1.23.5
+
+vmNamingTemplate: "oa93456{{.machine.Name |sha256sum | trunc 7}}"
```

Comment:
We can define this logic in the chart and consume only the prefix (e.g. `oa93456`) from user values. Nevertheless, users should be able to define their custom logic completely.

Reply:
Just saw this when passing by and want to leave my UX concerns here. Take it as my opinon.

Allowing users to specify arbitrary template code here is pretty challenging. It's highly likely that users make mistakes here.

Try to imagine a UI where users would enter this code: how could the UI assist to avoid mistakes? Can we validate the input? It's likely not easy.

If we cannot validate properly, we will submit unvalidated code that will fail somewhere in the cluster creation process. What will the roundtrip time be? Will the cluster come up in a broken state? Will this cause a debugging session.

So what would be the alternative?

- Have one optional sub property to allow to specify a prefix, e. g. `oa`.
- Have one optional sub property to specify whether the machine name should be included.
- Have one optional sub property to specify the length of the resulting name
- Always fill the maximum chars length with SHA256 checksum characters

Example:

```yaml
vmNamingRules:
  prefix: oa
  length: 7
  includeMachineName: true
```

Reply:
The example you used was one of our options when thinking about this at the beginning but we did it the way we did to make it as generic as possible so it can be merged upstream and avoid drifting further out from them. 

This feature will most likely be used only by us and the customer will always use the same template that we created for them so there shouldn't be any support action required there.

Reply:
I created this PR to just show how the upstream feature can be used on our side. I am not sure how generic the interface should be. We have to keep this repo customer agnostic so we cannot use this specific template for this repo as default. On the other hand, when we have the configuration interface so generic/customizable, it is error-prone. GitOps templates can be useful to define it once for a customer and reuse it for all WCs.